meeteval/der/md_eval.py

```diff
@@ -0,0 +1,175 @@
+        )
+
+
+def md_eval_22(
+        reference,
+        hypothesis,
+        average_out='{parent}/{stem}_md_eval_22.json',
+        per_reco_out='{parent}/{stem}_md_eval_22_per_reco.json',
+        collar=0,
+        regex=None,
+):
+    from meeteval.wer.__main__ import _load_texts
+
+    r, _, h, hypothesis_paths = _load_texts(
+        reference, hypothesis, regex)
+
+    r = _fix_channel(r.to_rttm())
+    h = _fix_channel(h.to_rttm())
+
+    r = r.grouped_by_filename()
+    h = h.grouped_by_filename()
+
+    keys = set(r.keys()) & set(h.keys())
+    missing = set(r.keys()) ^ set(h.keys())
+    if len(missing) > 0:
+        logging.warning(f'Missing {len(missing)} filenames:', missing)
+        logging.warning(f'Found {len(keys)} filenames:', keys)
+
+    md_eval_22 = shutil.which('md-eval-22.pl')
+    if not md_eval_22:
+        md_eval_22 = Path(__file__).parent / 'md-eval-22.pl'
+        if md_eval_22.exists():
+            pass
+        else:
+            url = 'https://github.com/nryant/dscore/raw/master/scorelib/md-eval-22.pl'
+            logging.info(f'md-eval-22.pl not found. Trying to download it from {url}.')
+            import urllib.request
+            urllib.request.urlretrieve(url, md_eval_22)
+            logging.info(f'Wrote {md_eval_22}')
+
+    def get_details(r, h, key, tmpdir):
+        r_file = tmpdir / f'{key}.ref.rttm'
+        h_file = tmpdir / f'{key}.hyp.rttm'
+        r.dump(r_file)
+        h.dump(h_file)
+
+        cmd = [
+            'perl', f'{md_eval_22}',
+            '-c', f'{collar}',
+            '-r', f'{r_file}',
+            '-s', f'{h_file}',
+        ]
+
+        cp = subprocess.run(cmd, stdout=subprocess.PIPE,
+                            check=True, universal_newlines=True)
+        # SCORED SPEAKER TIME =4309.340250 secs
+        # MISSED SPEAKER TIME =4309.340250 secs
+        # FALARM SPEAKER TIME =0.000000 secs
+        # SPEAKER ERROR TIME =0.000000 secs
+        #  OVERALL SPEAKER DIARIZATION ERROR = 100.00 percent of scored speaker time  `(ALL)
+
+        error_rate, = re.findall(r'OVERALL SPEAKER DIARIZATION ERROR = ([\d.]+) percent of scored speaker time', cp.stdout)
+        length, = re.findall(r'SCORED SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        deletions, = re.findall(r'MISSED SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        insertions, = re.findall(r'FALARM SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        substitutions, = re.findall(r'SPEAKER ERROR TIME =([\d.]+) secs', cp.stdout)
+
+        def convert(string):
+            return decimal.Decimal(string)
+
+        details = dict(
+            scored_speaker_time=convert(length),
+            # errors=float(errors),
+            missed_speaker_time=convert(deletions),
+            falarm_speaker_time=convert(insertions),
+            speaker_error_time=convert(substitutions),
+            error_rate=convert(error_rate) / 100,
+        )
+        return DiaErrorRate(**details)
```

Comment:
Why don't you construct the `DiaErrorRate` directly? The `details` dict seems redundant

Reply:
done